ROMFS/CMakeLists.txt

```diff
@@ -4,6 +4,22 @@ message(STATUS "ROMFS: ${config_romfs_root}")
 set(romfs_temp_dir ${PX4_BINARY_DIR}/ROMFS/${config_romfs_root})
 set(romfs_src_dir ${PX4_SOURCE_DIR}/ROMFS/${config_romfs_root})
 
+# Copy the optional files into ROMFS
+message(STATUS "The optional files: ${romfs_optional_files}")
+set(optional_files)
+foreach(optional_file ${romfs_optional_files})
+       get_filename_component(file_name ${optional_file} NAME)
+       set(file_dest ${romfs_temp_dir}/extras/${file_name})
+       add_custom_command(OUTPUT ${file_dest}
+               COMMAND mkdir -p ${romfs_temp_dir}/extras/
+               COMMAND cmake -E copy ${optional_file} ${file_dest}
+               DEPENDS ${optional_file}
+               COMMENT "Adding extra ROMFS file: ${optional_file}"
+               )
+       list(APPEND optional_files ${file_dest})
+endforeach()
+add_custom_target(collect_optional_extras DEPENDS ${optional_files})
```

Comment:
Doesn't the dependency need to be on a custom command OUTPUT to work properly? ${file_dest} in this case.

Reply:
@dagar. I am not sure... I know that if I don't have the `DEPENDS ${optional_files}`, then it doesn't include all of the additional files correctly. If I do `DEPENDS ${file_dest}`, it will only include the last file in the `romfs_optional_files` list in the ROMFS extra directory. I want the target `collect_optional_extras` 

From the CMake documentation on `add_custom_command`, it says the following:
>Dependencies listed with the DEPENDS argument may reference files and outputs of custom commands created with add_custom_command() in the same directory (CMakeLists.txt file).

I understood this to mean the DEPENDS argument ma reference file. Or it may reference outputs created with `add_custom_command()`. I didn't understand it to mean "files created with `add_custom_command()`". Admittedly, I don't fully understand the ins and outs of CMake, so I could well be wrong.

Reply:
I found this confusing as well. One thing you can do is generate with ninja, then in the build directory do `ninja -v -j1 -d explain`. After a full build you can touch an optional file, run the command, then see if the dependencies are working as expected (including actually rebuilding the final binary).

Reply:
@dagar: I think I have now established that my dependencies are not correct. After the first build, if I touch any of my extra files (either in the build directory or in the source directory) then the ROMFS and .px4 file are not being rebuilt.

I have tried for the last few hours to get this to work, but I am just going in circles. :-(

I have noticed that if I touch one of the normal ROMFS source files, e.g. any of the files in `Firmware/ROMFS/px4fmu_common/mixers` then the ROMFS and the .px4 file is also not rebuilt. I suspect that this is not the desired behaviour.

I also noticed that the variable `romfs_src_files` (one of the DEPENDS in ROMFS/CMakeLists.txt) isn't set anywhere, so I don't think it is doing anything.

Reply:
@dagar: Maybe I just needed to sleep on it. I think I have it working correctly now.... :-)